src/main/scala/beam/utils/ScenarioReaderCsv.scala

```diff
@@ -0,0 +1,378 @@
+import scala.collection.concurrent.TrieMap
+import scala.collection.mutable.ListBuffer
+import scala.collection.parallel.mutable.ParTrieMap
+
+class ScenarioReaderCsv(var scenario: MutableScenario, var beamServices: BeamServices, val delimiter: String = ",")
+    extends LazyLogging {
+
+  val scenarioFolder = beamServices.beamConfig.beam.agentsim.agents.population.beamPopulationDirectory
+
+  val buildingFilePath = scenarioFolder + "/buildings.csv"
+  val personFilePath = scenarioFolder + "/persons.csv"
+  val householdFilePath = scenarioFolder + "/households.csv"
+
+  val planFilePath = scenarioFolder + "/plans.csv"
+  val unitFilePath = scenarioFolder + "/units.csv"
+  val parcelAttrFilePath = scenarioFolder + "/parcel_attr.csv"
+
+  def loadScenario() = {
+
+    scenario.getPopulation.getPersons.clear()
+    scenario.getPopulation.getPersonAttributes.clear()
+    scenario.getHouseholds.getHouseholds.clear()
+    scenario.getHouseholds.getHouseholdAttributes.clear()
+
+    beamServices.vehicles.clear()
+    beamServices.privateVehicles.clear()
+    /////
+
+    logger.info("Reading units...")
+    val units = ScenarioReaderCsv.readUnitsFile(unitFilePath)
+
+    logger.info("Reading parcel attrs")
+    val parcelAttrs = ScenarioReaderCsv.readParcelAttrFile(parcelAttrFilePath)
+
+    logger.info("Reading Buildings...")
+    val buildings = ScenarioReaderCsv.readBuildingsFile(buildingFilePath)
+
+    logger.info("Reading Persons...")
+    val householdPersons = ScenarioReaderCsv.readPersonsFile(
+      personFilePath,
+      scenario.getPopulation,
+      BeamMode.allBeamModes.map(_.value).mkString(",")
+    )
+
+    logger.info("Reading plans...")
+    ScenarioReaderCsv.readPlansFile(planFilePath, scenario.getPopulation, beamServices)
+
+    logger.info("Total persons loaded {}", scenario.getPopulation.getPersons.size())
+    logger.info("Checking persons without selected plan...")
+
+    val listOfPersonsWithoutPlan: ListBuffer[Id[Person]] = ListBuffer()
+    scenario.getPopulation.getPersons.forEach {
+      case (pk: Id[Person], pv: Person) if (pv.getSelectedPlan == null) => listOfPersonsWithoutPlan += pk
+      case _                                                            =>
+    }
+
+    logger.info("Removing persons without plan {}", listOfPersonsWithoutPlan.size)
+    listOfPersonsWithoutPlan.foreach { p =>
+      val _hId = scenario.getPopulation.getPersonAttributes.getAttribute(p.toString, "householdId")
+
+      scenario.getPopulation.getPersonAttributes.removeAllAttributes(p.toString)
+
+      scenario.getPopulation.removePerson(p)
+
+      val hId = Id.create(_hId.toString, classOf[Household])
+      householdPersons.get(hId) match {
+        case Some(persons: ListBuffer[Id[Person]]) =>
+          persons -= p
+        case None =>
+      }
+    }
+
+    logger.info("Reading Households...")
+    ScenarioReaderCsv.readHouseholdsFile(
+      householdFilePath,
+      scenario,
+      beamServices,
+      householdPersons.par,
```

Comment:
Why make it as a parallel collection here? I don't see any `map` or other functions on these collections. I can see that it's used only for lookup (`.get`), so it won't give any benefit.